src/components/common/CreateLocation.js

```diff
@@ -0,0 +1,143 @@
+import React, {useState} from 'react';
+import Modal from 'react-bootstrap/Modal';
+import PrimaryButton from "../common/PrimaryButton";
+import Form from 'react-bootstrap/Form';
+import FormControl from 'react-bootstrap/FormControl';
+import SecondaryButton from './SecondaryButton';
+
+const CreateLocation = ({ show, handleClose }) => {
+    const [formData, setFormData] = useState([
+        {id: 1, label: 'Title', type: 'text', value: ''},
+        {id: 2, label: 'Latitude', type: 'number', value: ''},
+        {id: 3, label: 'Longitude', type: 'number', value: ''},
+        {id: 4, label: 'Type', type: 'checkbox', value: '', options: ['Gym-Lead', 'Outdoor-Boulder', 'Gym-Boulder']},
+        {id: 5, label: 'Experience Level', type: 'radio', value: '', options: ['Beginner', 'Intermediate', 'Advanced']},
+        {id: 6, label: 'Description', type: 'text', value: ''},  
+        {id: 7, label: 'Picture', type: 'file', value: ''},
+    ]);
+
+    const handleInputChange = (id, e) => {
+        const updatedFormData = formData.map(item => 
+            item.id === id ? {...item, value: e.target.value} : item
+        );
+        setFormData(updatedFormData);
+    }
+
+    const handleSubmit = (e) => {
+        e.preventDefault();
+        //code for submitting data to API
+    }
+   
+  return (
+    <>
+      <Modal show={show} onHide={handleClose}>
+        <Modal.Header closeButton>
+          <Modal.Title>Create Location</Modal.Title>
+        </Modal.Header>
+        <Modal.Body>
+            <Form onSubmit={handleSubmit}>
+                <div className="row w-100">
+                        <Form.Group controlId="formControl_1">
+                            <Form.Label>Title</Form.Label>
+                                <FormControl
+                                    type="text"
+                                    placeholder="Enter Title"
+                                    value={formData[0].value}
```

Comment:
formData[0] - is hard to read. 
either do a map over your list -- which is advanced and great; 
or just define multiple state variables, such that the code is readable. 
but this is a strange compromise.